test_programs/compile_success_empty/serialize/src/main.nr

```diff
@@ -0,0 +1,59 @@
+trait Serialize {
+    let Size: u32;
+
+    // Note that Rust disallows referencing constants here!
+    fn serialize(self) -> [Field; Self::Size];
+}
+
+impl<A, B, let AS: u32, let BS: u32> Serialize for (A, B) where A: Serialize<Size = AS>, B: Serialize<Size = BS> {
+    // let Size = <A as Serialize>::Size + <B as Serialize>::Size;
+    let Size = AS + BS;
+
+    fn serialize(self: Self) -> [Field; Self::Size] {
+        let mut array: [Field; Self::Size] = std::mem::zeroed();
+        let a = self.0.serialize();
+        let b = self.1.serialize();
+
+        for i in 0 .. a.len() {
+            array[i] = a[i];
+        }
+        for i in 0 .. b.len() {
+            array[i + a.len()] = b[i];
+        }
+        array
+    }
+}
+
+impl<T, let N: u32, let TS: u32> Serialize for [T; N] where T: Serialize<Size = TS> {
+    // let Size = <T as Serialize>::Size * N;
```

Comment:
I see this commented out here as well? Should we leave a todo note?

Reply:
I've made an issue for it. I'll probably split it into a separate test once it is implemented so that we can continue to check both options.